pkg/objectwait/rte/rte.go

```diff
@@ -94,9 +114,64 @@ func Deletable(mf rtemf.Manifests, cli client.Client, log logr.Logger) []objectw
 	}
 	if mf.MachineConfig != nil {
 		objs = append(objs, objectwait.WaitableObject{
-			// TODO: we should add functionality to wait for the MCP update
 			Obj: mf.MachineConfig,
+			Wait: func(ctx context.Context) error {
+				mcps, err := getMPCsForMC(cli, ctx, *mf.MachineConfig)
+				if err != nil {
+					return err
+				}
+				pollInterval := 30 * time.Second
+				pollTimeout := 30 * time.Minute
+				err = waitForMachineConfigPoolsCondition(cli, log, ctx, mcps, mcov1.MachineConfigPoolUpdating, pollInterval, pollInterval)
+				if err != nil {
+					return err
+				}
+				return waitForMachineConfigPoolsCondition(cli, log, ctx, mcps, mcov1.MachineConfigPoolUpdated, pollInterval, pollTimeout)
+			},
 		})
 	}
 	return objs
 }
+
+func waitForMachineConfigPoolsCondition(cli client.Client, log logr.Logger, ctx context.Context, mcps []*mcov1.MachineConfigPool, cond mcov1.MachineConfigPoolConditionType, interval, timeout time.Duration) error {
```

Comment:
this function could fit in the `wait` package, perhaps later.
The biggest problem however is we don't have OCP in our CI yet (and it's hard to get fixed) so we either add robust unit test coverage, we start using mock aggressively or we postpone the additions of more OCP-specific code which isn't quite trivial.

I'm fine with any of the above approaches.

Reply:
I would like to give it a try to the test coverage.
I am not sure though if there is a mock framework of preference